common:
    lid_topic:  "/velodyne_points_revise"
    imu_topic:  "/quadrotor_simulator_so3_2/imu"
    points_topic: "/velodyne_points"
    odom_topic: "//odometry_gt_"
    odom_out_topic: "/aft_mapped_to_init"
    lid_out_topic: "/cloud_registered_body"
    time_sync_en: true         # ONLY turn on when external time synchronization is really not possible

dyn_obj:
    dataset: 0 #0 for kitti, 1 for nuscenes, 2 for waymo
    use_odom_file: true
    buffer_delay: 0.1
    buffer_size: 300000
    points_num_perframe: 150000
    depth_map_dur: 0.1
    dyn_windows_dur: 0.1
    max_depth_map_num: 5
    dyn_windows_num: 5
    max_pixel_points: 50
    frame_dur: 0.1
    self_x_f: 2.5
    self_x_b: -1.5
    self_y_l: 0 #1.6
    self_y_r: -0 #1.6
    blind_dis: 0.3
    fov_up: 2.0
    fov_down: -23.0
    fov_left: 180.0
    fov_right: -180.0
    fov_cut: -20
    checkneighbor_range: 1
    
    depth_thr1: 0.15
    enter_min_thr1: 1.0 # kill car front points (close to wall, related to view change)
    enter_max_thr1: 0.3 # equal to map_cons_depth_thr1, for save pedestrain close to wall
    map_cons_depth_thr1: 0.3
    map_cons_hor_thr1: 0.01
    map_cons_ver_thr1: 0.015
    map_cons_hor_dis1: 0.15
    map_cons_ver_dis1: 0.15
    depth_cons_depth_thr1: 0.5
    depth_cons_depth_max_thr1: 1.0
    depth_cons_hor_thr1: 0.01
    depth_cons_ver_thr1: 0.01
    enlarge_z_thr1 : -2.5
    enlarge_angle: 2.0
    enlarge_depth: 3.0
    occluded_map_thr1: 3
    occlude_windows: 3
    case1_interp_en: true

    v_min_thr2: 0.5
    acc_thr2: 7.0
    map_cons_depth_thr2: 0.2
    map_cons_hor_thr2: 0.01
    map_cons_ver_thr2: 0.01
    occ_depth_thr2: 10.15
    occ_hor_thr2: 0.01
    occ_ver_thr2: 0.015
    depth_cons_depth_thr2: 0.1
    depth_cons_depth_max_thr2: 0.5
    depth_cons_hor_thr2: 0.01
    depth_cons_ver_thr2: 0.01
    occluded_times_thr2: 3
    k_depth2: 0.005
    case2_interp_en: false

    v_min_thr3: 0.5
    acc_thr3: 15.0
    map_cons_depth_thr3: 0.2
    map_cons_hor_thr3: 0.01
    map_cons_ver_thr3: 0.008
    occ_depth_thr3: 0.15
    occ_hor_thr3: 0.01
    occ_ver_thr3: 0.01
    depth_cons_depth_thr3: 0.3
    depth_cons_depth_max_thr3: 1.0
    depth_cons_hor_thr3: 0.01
    depth_cons_ver_thr3: 0.01
    occluded_times_thr3: 3
    case3_interp_en: false

    interp_hor_thr: 0.015
    interp_ver_thr: 0.007
    interp_thr1: 0.5
    interp_static_max: 10.0
    interp_start_depth1: 10
    interp_kp1: 0.1
    interp_kd1: 0.5
    interp_bg: 0.8
    interp_thr2: 0.25
    interp_thr3: 0.05

    point_index:  -1 #20751 #7232 #17929 #24784
    debug_x: 7.2065
    debug_y: -3.1844
    debug_z: -0.89005
    cluster_coupled: true
    cluster_future: true
    cluster_extend_pixel: 2
    cluster_min_pixel_number: 8
    cluster_thrustable_thresold: 0.3
    cluster_Voxel_revolusion: 0.3
    # pose_file: "/home/huajie/event_detection/LiDAR-MOS/data/sequences/00/poses.txt"
    # out_file: "/home/huajie/event_detection/kitti/sequences/20/predictions/"
    # out_file: "/home/huajie/event_detection/bag/revise/sequences/20/predictions/"
    # pc_file: "/home/huajie/event_detection/LiDAR-MOS/data/sequences/00/velodyne/"predictions_1
    label_file: "/home/huajie/event_detection/kitti/dataset/labels/0015/"
    pred_file: "/home/huajie/event_detection/LiDAR-MOS/data/smos_kitti/predict_smos/0015/"
    # pred_file: "/home/huajie/event_detection/kitti/residual_8_semantic/45/predictions/"
    # pred_file: "/home/huajie/event_detection/kitti/dataset/predictions_kitti7/0015/"
    # pred_file: "/home/huajie/event_detection/kitti/sequences/20/predictions/"

    # pred_file: "/home/huajie/event_detection/LiDAR-MOS/data/salsanext_residual_8_sem_allseqs/sequences/21/predictions/"

    # pose_file: "/media/yihang/LYH/kitti/dataset/sequences/13/KITTI_13.txt"
    # out_file: &out_file_path "/media/yihang/LYH/kitti/dataset/sequences/20/0927(fixbug)/"
    # # pc_file : "/media/yihang/LYH/kitti/dataset/sequences/08/velodyne/"
    # label_file : "/media/yihang/LYH/kitti/dataset/sequences/20/0923(7map)/"
    # label_file : "/media/yihang/LYH/kitti/salsanext_residual_8_sem_allseqs/sequences/20/predictions/"
    # # pred_file : "/media/yihang/LYH/kitti/predictions_salsanext_semantic/salsanext_semantic/sequences/20/predictions/"
    # pred_file : *out_file_path
    # semantic_file: "/media/yihang/LYH/kitti/predictions_salsanext_semantic/salsanext_semantic/sequences/21/predictions/"
    # pred_semantic_file: "/media/yihang/LYH/kitti/submit/0603_semantic/sequences/11/---/"

    #for kitti tracking dataset
    # pose_log: false
    # # pose_log_file: "/media/yihang/LYH/kitti_tracking/data_tracking_dataset/training/pose_lio/0000.txt"
    # pose_log_file: "/media/yihang/LYH/kitti/dataset/sequences/10/10_static.txt"
    # out_file: "/media/yihang/LYH/kitti_tracking/data_tracking_dataset/training/predict_points/0000/"
    # pc_file : "/media/yihang/LYH/kitti_tracking/data_tracking_dataset/training/velodyne/0009/"
    # label_file : "/media/yihang/LYH/kitti_tracking/data_tracking_dataset/training/label_points_recall/0020/"
    # pred_file : "/media/yihang/LYH/kitti_tracking/data_tracking_dataset/training/predict_points/0020/"
    # bin_folder: "/media/yihang/LYH/kitti_tracking/data_tracking_dataset/training/velodyne_dyn_revise/0020/"

    #for semantic
    # pred_file : "/home/huajie/event_detection/kitti/residual_8/50/predictions/"
    # semantic_file: "/home/huajie/event_detection/kitti/semantic/50/predictions/"
    # pred_semantic_file: "/home/huajie/event_detection/kitti/residual_8_semantic/50/predictions/"

    # dataset_folder: "/home/huajie/event_detection/kitti/"


    mode: 0 # 0 for distance resolution and 1 for angle resolution
    dyn_filter_en: false
    debug_publish: false
    
    interp_en: false
    is_add: false
    quadtree_layer_max: 4 ###
    ver_resolution_max: 0.007
    hor_resolution_max: 0.005
    angle_noise: 0.02 #couple with occu times
    
    min_dis_roll1: 0.3
    min_dis_pitch1: 0.1
    min_roll1: 0.02
    min_pitch1: 0.012
    depth_diff: 0.2
    depth_cut: 10.0
    roll_noise: 0.01
    pitch_noise: 0.005
    min_dis_roll2: 0.2
    min_dis_pitch2: 0.05
    min_roll2: 0.01
    min_pitch2: 0.005
    depth2: 0.5
    v_min_case2: 0.5
    depth_thr_case2: 0.15
    depth_minus: 0.5
    
    angle_occlude: 0.02
    depth_thr: 0.15
    depth_thr_fpc_max: 0.5
    depth_thr_fpc_min: 0.1


    
preprocess:
    lidar_type: 4                # 1 for Livox serials LiDAR, 2 for Velodyne LiDAR, 3 for ouster LiDAR, 4 for Realsense L515
    feature_extract_en: false
    scan_line: 6
    blind: 0

voxel:
    voxel_map_en: false
    pub_plane_en: false
    match_eigen_value: 0.005
    min_eigen_value: 0.005
    match_s: 0.9
    sigma_num: 81
    voxel_size: 3
    layer: 2

mapping:
    imu_en: false
    init_vel_x: 10.0
    init_vel_y: 0.0
    init_vel_z: 0.0
    acc_cov: 0.5
    gyr_cov: 0.2
    b_acc_cov: 0.0001
    b_gyr_cov: 0.0001
    fov_degree:    180
    det_range:     100.0
    extrinsic_T: [ 0.0, 0.0, -0.0 ]
    extrinsic_R: [ 1, 0, 0,
                   0, 1, 0,
                   0, 0, 1]

publish:
    path_en:  false
    scan_publish_en:  true       # false: close all the point cloud output
    dense_publish_en: true       # false: low down the points number in a global-frame point clouds scan.
    scan_bodyframe_pub_en: false  # true: output the point cloud scans in IMU-body-frame

pcd_save:
    pcd_save_en: false
    interval: -1                 # how many LiDAR frames saved in each pcd file; 
                                 # -1 : all frames will be saved in ONE pcd file, may lead to memory crash when having too much frames.